### PR TITLE
[5.2.0] Update java_tools 11.7.1

### PR DESCRIPTION
**Pull request by hvadehra**
(cherry picked from commit https://github.com/bazelbuild/bazel/commit/22c91dab43558503a1d6759ccfd77587d87b28cd)

### CODE REVIEW

**Comment by meteorcloudy** [comment]
Please wait on merging this since this change actually breaks JDK8 compatibility.

**Comment by hvadehra** [comment]
@coeuvre all failures have now been fixed, so this can be merged now.

**Comment by meteorcloudy** [comment]
@hvadehra Can you fix the CLA error?

**Comment by hvadehra** [comment]
> @hvadehra Can you fix the CLA error?

@coeuvre Done

File: src/main/java/com/google/devtools/build/lib/analysis/BuildView.java [hunk 1]
```diff
@@ -410,7 +410,7 @@ public AnalysisResult update(
               "Analysis succeeded for only %d of %d top-level targets",
               numSuccessful, numTargetsToAnalyze);
       eventHandler.handle(Event.info(msg));
-      logger.atInfo().log(msg);
+      logger.atInfo().log("%s", msg);
```
Comment (meteorcloudy) [inline src/main/java/com/google/devtools/build/lib/analysis/BuildView.java:413]: What's up with those changes? I don't see them in https://github.com/bazelbuild/bazel/commit/22c91dab43558503a1d6759ccfd77587d87b28cd

**Review by meteorcloudy — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/devtools/build/lib/analysis/BuildView.java (hunk 1 above)
Comment (hvadehra) [reply src/main/java/com/google/devtools/build/lib/analysis/BuildView.java:413]: There are two more (partial) cherry picks from: 1987af2216b3adb7d0e19a70f0e1778eee519d4c and ccc9b9d65623d454d987185af4e14ce26a67fa53 to fix the error prone failures.

**Review by meteorcloudy — APPROVED** [review]

**Pull request merged**